src/dos/program_more.cpp

```diff
@@ -0,0 +1,508 @@
+		WriteOut("\n");
+		++line_counter;
+	} else {
+		// We are going to continue - erase the prompt
+		WriteOut("\033[M"); // clear line
+		auto counter = GetCurrentColumn();
+		while (counter--)
+			WriteOut("\033[D"); // cursor one position back
+	}
+
+	return decision;
+}
+
+bool MORE::GetCharacter(char &code)
+{
+	// clang-format off
+	constexpr std::array<char, 7> break_sequence = {
+		// CTRL + ZXCVBNM - emergency break key sequence, in case
+	    // user executes MORE.COM with STDIN as input; supported
+	    // at least by MS-DOS 6.22, DR-DOS 7.03 and PC-DOS 2000
+	    0x1a, 0x18, 0x03, 0x16, 0x02, 0x0e, 0x0d
+	};
+	// clang-format on
+
+	if (!tab_left) {
+		while (true) {
+			// Retrieve character from input stream
+			uint16_t count = 1;
+			DOS_ReadFile(input_handle,
+			             reinterpret_cast<uint8_t *>(&code),
+			             &count);
+			if (!count) {
+				break_idx = 0;
+				return false; // end of stream
+			}
+
+			// Handle emergency break sequence
+			if (break_enable && (code == break_sequence[break_idx])) {
+				if (++break_idx == break_sequence.size()) {
+					break_idx = 0;
+					return false; // emergency break
+				}
+			} else
+				break_idx = 0;
+
+			// Skip CR/LF characters if requested
+			if (skip_next_cr && code == code_cr)
+				skip_next_cr = false;
+			else if (skip_next_lf && code == code_lf)
+				skip_next_lf = false;
+			else
+				break;
+		}
+
+		// If TAB found, replace it with given number of spaces
+		if (code == '\t')
+			tab_left = tab_size;
+	}
+
+	if (tab_left) {
+		--tab_left;
+		code = ' ';
+	}
+
+	return true;
+}
+
+uint8_t MORE::GetCurrentColumn()
+{
+	const auto page = real_readb(BIOSMEM_SEG, BIOSMEM_CURRENT_PAGE);
+	return CURSOR_POS_COL(page);
+}
+
+uint8_t MORE::GetCurrentRow()
+{
+	const auto page = real_readb(BIOSMEM_SEG, BIOSMEM_CURRENT_PAGE);
+	return CURSOR_POS_ROW(page);
+}
+
+void MORE::AddMessages()
+{
+	MSG_Add("SHELL_CMD_MORE_HELP",
+	        "Display command output or text file one screen at a time.\n");
+	MSG_Add("SHELL_CMD_MORE_HELP_LONG",
+	        "Display command output or text file one screen at a time.\n"
+	        "\n"
+	        "Usage:\n"
+	        "  [color=cyan]COMMAND[reset] | [color=green]more[reset] [/t[color=white]n[reset]]\n"
+	        "  [color=green]more[reset] [/t[color=white]n[reset]] < [color=cyan]FILE[reset]\n"
+	        "  [color=green]more[reset] [/t[color=white]n[reset]] [color=cyan]PATTERN[reset] [[color=cyan]PATTERN[reset] ...]\n"
+	        "\n"
+	        "Where:\n"
+	        "  [color=cyan]COMMAND[reset] is the command to display output one screen at a time.\n"
+	        "  [color=cyan]FILE[reset]    is an exact filename of the file to display.\n"
+	        "  [color=cyan]PATTERN[reset] is either an exact filename or an inexact filename with wildcards,\n"
+	        "          which are the asterisk (*) and the question mark (?).\n"
+	        "  [color=white]n[reset]       is a tabulation size, 1-9, default is 8.\n"
```

Comment:
> `"is a tabulation size"`

"To tabulate" is somewhat of data organization term of arranging information in column format. "Let me tabulate the census records".

Although it can be used, I would suggest tab-size as a more common term used in text processing tools and editors. 

Suggest:

`"is the tab-size, which specifies the width of a tab character. The  defaults is 8, but can be set from 1 to 9."`

Reply:
Changed to _tab size_.